frontend-react/src/shared/Tooltip.test.tsx

```diff
@@ -0,0 +1,113 @@
+import { describe, test } from "@jest/globals";
+import { screen } from "@testing-library/react";
+import React from "react";
+import user from "@testing-library/user-event";
+
+import { renderApp } from "../utils/CustomRenderUtils";
+
+import {
+    Tooltip,
+    useTooltipContext,
+    TooltipContext,
+    TooltipContextType,
+    TooltipTrigger,
+    TooltipContent,
+} from "./Tooltip";
+
+const TooltipContextReceiverComponent = () => {
+    const { placement } = useTooltipContext();
+    return <>{placement}</>;
+};
+
+const TooltipContextCreatorComponent = ({
+    children,
+}: {
+    children: React.ReactNode;
+}) => {
+    const ctx: TooltipContextType = {
+        placement: "Test",
+    } as any;
+    return (
+        <TooltipContext.Provider value={ctx}>
+            {children}
+        </TooltipContext.Provider>
+    );
+};
+
+describe("Tooltip", () => {
+    describe("useTooltipContext", () => {
+        test("no context - error", () => {
+            expect(() =>
+                renderApp(<TooltipContextReceiverComponent />)
+            ).toThrowError("Tooltip components must be wrapped in <Tooltip />");
+        });
+
+        test("context", () => {
+            renderApp(
```

Comment:
Along the same lines of the previous point, I think `renderHook` also takes a `wrapper`.  Either way works, though!  Definitely not a big deal!